pyfar/dsp/filter/gammatone.py

```diff
@@ -0,0 +1,529 @@
+import numpy as np
+import scipy.signal as sgn
+import pyfar as pf
+
+
+class GammatoneBands():
+    """
+    Generate reconstructing auditory filter bank.
+
+    Generate a forth order reconstructing Gammatone auditory filter bank
+    according to [#]_. This is a Python port of the `hohmann2002` filter bank
+    contained in the Auditory Modeling Toolbox [#]_. The filter bank can
+    handle single and multi channel input and allows for an almost perfect
+    reconstruction of the input signal (see examples below).
+
+    Calling ``GFB = GammatoneBands()`` constructs the filter bank. Afterwards
+    the class methods ``GFB.filter()`` and ``GFB.sum`` can be used to filter
+    and reconstruct signals. All relevant data such as the filter coefficients
+    can be obtained for example through ``GFB.coefficients``. See below fore
+    more documentation.
+
+    Parameters
+    ----------
+    freq_range : array like
+        The upper and lower frequency in Hz between which the filter bank is
+        constructed. Values must be larger than 0 and not exceed half the
+        sampling rate.
+    resolution : number
+        The frequency resolution of the filter bands in equivalent rectangular
+        bandwidth (ERB) units. The bands of the filter bank are distributed
+        linearly on the ERB scale. The default value of ``1`` results in one
+        filter band per ERB. A value of ``0.5`` would result in two filter
+        bands per ERB.
+    reference_frequency : number
+        The frequency relative to which the filter bands are distributed. The
+        default is ``1000`` Hz.
+    delay : number
+        The delay in seconds that the filter bank will have, i.e., the delay
+        that is added to the input signal after being filtered and summed
+        again. The default is ``0.004`` seconds.
```

Comment:
Not knowing much about the gammatone filterbank: Is 0.004 some kind of standard value or where does it come from?